### PR TITLE
[WebLocks]: Specifying ordering for navigator.locks.query()

### PR DESCRIPTION
**Pull request by andreas-butler**
Adds a paragraph explaining the ordering requirement for the state snapshot's ordering.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -252,21 +252,22 @@ A <dfn>lock request</dfn> represents a pending request for a [=lock-concept|lock
 
 A [=lock request=] is a tuple of (<dfn>agent</dfn>, <dfn>clientId</dfn>, <dfn>origin</dfn>, <dfn>name</dfn>, <dfn>mode</dfn>, <dfn>promise</dfn>).
 
-Each origin has an associated <dfn>lock request queue</dfn>, which is a [=queue=] of [=lock requests=].
+Each origin has an associated <dfn>lock request queue set</dfn>, which is a [=set=] of [=lock request queues=] (a [=lock request queue=] being a [=queue=] of [=lock requests=]) where each queue is associated with a particular resource name.
```
Comment (inexorabletash) [inline index.bs]: This sounds like a [map](https://infra.spec.whatwg.org/#maps) ? That would formalize the key/value nature (key being the name, value being the lock request queue)?

This needs a `<dfn>` for _lock request queue_, and that should be moved to its own line. Defining it in () makes the sentence hard to parse.

File: index.bs [hunk 2]
```diff
@@ -252,21 +252,22 @@ A <dfn>lock request</dfn> represents a pending request for a [=lock-concept|lock
...
 
 A [=lock request=] |request| is said to be <dfn>grantable</dfn> if the following steps return true:
 
 1. Let |origin| be |request|'s [=lock request/origin=].
-1. Let |queue| be |origin|'s [=lock request queue=]
+1. Let |queue set| be |origin|'s [=lock request queue set=]
+1. Let |name| be |request|'s associated [=lock request/name=]
+1. Let |queue| be |name|'s associated [=lock request queue=] in |queue set|
```
Comment (inexorabletash) [inline index.bs]: This gets more formal using [maps](https://infra.spec.whatwg.org/#maps)

File: index.bs [hunk 3]
```diff
@@ -252,21 +252,22 @@ A <dfn>lock request</dfn> represents a pending request for a [=lock-concept|lock
...
 1. Let |held| be |origin|'s [=held lock set=]
 1. Let |mode| be |request|'s associated [=lock request/mode=]
-1. Let |name| be |request|'s associated [=lock request/name=]
 1. If |mode| is "{{exclusive}}", then return true if all of the following conditions are true, and false otherwise:
     * No [=lock-concept|lock=] in |held| has a [=lock-concept/name=] that equals |name|
-    * No [=lock request=] in |queue| earlier than |request| has a [=lock request/name=] that equals |name|.
```
Comment (inexorabletash) [inline index.bs]: This gets simpler, yay!

File: index.bs [hunk 4]
```diff
@@ -512,7 +513,9 @@ The <dfn method for=LockManager>request(|name|, |callback|)</dfn> and
...
+    For any given resource, the snapshot of the pending lock requests should return the requests in the order in which they were made; however, no guarantees are made with respect to the relative ordering of requests across different resources. For example, if pending lock requests A1 and A2 are made against resource A in that order, and pending lock requests B1 and B2 are made against resource B in that order, than both {A1, A2, B1, B2} and {A1, B1, A2, B2} would be valid orderings for a snapshot's pending list, but neither of {A1, A2, B2, B1} or {B1, A2, B2, A1} would be valid (note these examples are not exhaustive).
```
Comment (inexorabletash) [inline index.bs]: We may want to pull this level of detail out of the "domintro" section into a separate note below the algorithm; the "domintro" section is intended to be quick documentation for developers. This detail is great, but not necessarily at the "introductory" level. But you can leave it here if you want and I can edit after merging.

File: index.bs [hunk 5]
```diff
@@ -631,20 +636,24 @@ To <dfn>abort the request</dfn> |request|:
 
 1. [=Assert=]: these steps are running on the [=lock task queue=].
 1. Let |origin| be |request|'s [=lock request/origin=].
-1. [=list/Remove=] |request| from |origin|'s [=lock request queue=].
-1. [=Process the lock request queue=] for |origin|.
+1. Let |name| be |lock|'s associated [=resource name=].
+1. Let |queue set| be |origin|'s [=lock request queue set=].
+1. Let |queue| be |name|'s associated [=lock request queue=] in |queue set|.
+1. [=list/Remove=] |request| from |queue|.
+1. [=Process the lock request queue=] |queue|.
```
Comment (inexorabletash) [inline index.bs]: need to pass in |origin| as an argument.

File: index.bs [hunk 6]
```diff
@@ -617,8 +621,9 @@ To <dfn>release the lock</dfn> |lock|:
 
 1. [=Assert=]: these steps are running on the [=lock task queue=].
 1. Let |origin| be |lock|'s [=lock-concept/origin=].
+1. Let |name| be |lock|'s associated [=resource name=].
 1. [=list/Remove=] [=lock-concept|lock=] from the |origin|'s [=held lock set=].
-1. [=Process the lock request queue=] for |origin|.
+1. [=Process the lock request queue=] associated with |name| in the |origin|'s [=lock request queue set=].
```
Comment (inexorabletash) [inline index.bs]: Ideally this would get |queue|, then invoke the algorithm with |queue| and |origin|

File: index.bs (hunk 5 above)
Comment (andreas-butler) [reply index.bs]: Looks like the origin is exposed through |request|. Is the suggestion to pass it in for clarity?

File: index.bs (hunk 3 above)
Comment (andreas-butler) [reply index.bs]: Sorry, I'm not sure I see how this gets much simpler?

File: index.bs (hunk 5 above)
Comment (inexorabletash) [reply index.bs]: As currently written, the "process the lock request queue" algorithm takes |name| and |origin| as arguments, so you need to "pass those in" or restructure the algorithm's "method signature". 

I agree that now it doesn't need |origin| passed in, so you could remove that from the "method signature", i.e. remove the "in |origin|" phrase.

File: q [hunk 7]
```diff
@@ -0,0 +1,1155 @@
+[33mcommit b2e4a2d084215d2a8b8cd530216e2ea2b651db6c[m[33m ([m[1;36mHEAD -> [m[1;32mquery-order-change[m[33m)[m
```
Comment (inexorabletash) [inline q]: This file is extraneous - remove it from the change?

File: index.bs (hunk 3 above)
Comment (inexorabletash) [reply index.bs]: Apologies - as mentioned offline, I was just happy that some text was deleted from the spec. Less words = better!

File: index.bs (hunk 5 above)
Comment (andreas-butler) [reply index.bs]: Ah, sorry, I thought you meant that the 'abort the request' algorithm needed to take |origin| as an argument. I see what you mean now.

File: index.bs [hunk 8]
```diff
@@ -252,21 +252,24 @@ A <dfn>lock request</dfn> represents a pending request for a [=lock-concept|lock
 
 A [=lock request=] is a tuple of (<dfn>agent</dfn>, <dfn>clientId</dfn>, <dfn>origin</dfn>, <dfn>name</dfn>, <dfn>mode</dfn>, <dfn>promise</dfn>).
 
-Each origin has an associated <dfn>lock request queue</dfn>, which is a [=queue=] of [=lock requests=].
+A <dfn>lock reqeust queue</dfn> is a [=queue=] of [=lock requests=].
```
Comment (inexorabletash) [inline index.bs]: typo: "reqeust" &rarr; "request"

File: index.bs [hunk 9]
```diff
@@ -669,12 +678,13 @@ To <dfn>snapshot the lock state</dfn> for |origin| with |promise|:
 
 1. [=Assert=]: these steps are running on the [=lock task queue=].
 1. Let |pending| be a new [=list=].
-1. [=list/For each=] |request| of |origin|'s [=lock request queue=]:
-    1. Let |info| be a new {{LockInfo}} dictionary.
-    1. Set |info|'s {{LockInfo/name}} dictionary member to |request|'s [=lock request/name=].
-    1. Set |info|'s {{LockInfo/mode}} dictionary member to |request|'s [=lock request/mode=].
-    1. Set |info|'s {{LockInfo/clientId}} dictionary member to |request|'s [=lock request/clientId=].
-    1. [=list/Append=] |info| to |pending|.
+1. [=list/For each=] |queue| of |origin|'s [=lock request queue set=]:
```
Comment (inexorabletash) [inline index.bs]: "lock request queue set" is no longer a thing. This will need to be "lock request queue map" and iterate using https://infra.spec.whatwg.org/#map-iterate

File: index.bs [hunk 10]
```diff
@@ -233,7 +233,7 @@ The [=lock-concept/waiting promise=] is not named in the above code, but is stil
 Further note that if the callback is not `async` and returns a non-promise, the return value is wrapped in a promise that is immediately resolved; the lock will be released in an upcoming microtask, and the [=lock-concept/released promise=] will also resolve in a subsequent microtask.
 </aside>
 
-Each origin has an associated <dfn>held lock set</dfn> which is an [=/ordered set=] of [=lock-concept|locks=].
+Each origin has an associated <dfn>held lock set</dfn> which is a [=set=] of [=lock-concept|locks=].
```
Comment (inexorabletash) [inline index.bs]: No way for you to know this without running bikeshed but: we'll need to use `[=/set=]` to disambiguate the term "set".

File: q (hunk 7 above)
Comment (inexorabletash) [reply q]: remember to `git rm` this file?
